tianshou/data/collector.py

```diff
@@ -145,26 +154,12 @@ def reset(
     ) -> None:
         """Reset the environment, statistics, current data and possibly replay memory.
 
-        :param reset_buffer: if true, reset the replay buffer that is attached
+        :param reset_buffer: if true, reset the replay buffer attached
             to the collector.
         :param gym_reset_kwargs: extra keyword arguments to pass into the environment's
             reset function. Defaults to None (extra keyword arguments)
         """
-        # use empty Batch for "state" so that self.data supports slicing
-        # convert empty Batch to None when passing data to policy
-        data = Batch(
-            obs={},
-            act={},
-            rew={},
-            terminated={},
-            truncated={},
-            done={},
-            obs_next={},
-            info={},
-            policy={},
-        )
-        self.data = cast(RolloutBatchProtocol, data)
-        self.reset_env(gym_reset_kwargs)
+        self._last_obs, self._last_info = self.reset_env(gym_reset_kwargs)
```

Comment:
rename to `_reset_obs` and `_reset_info`?

Do we generally need to reset in the constructor at all? Could we reset in `.collect`?

Reply:
_reset_obs could be easily taken as the obs to which to reset the env to. 
How about `_obs_from_reset` and `_info_from_reset`?

For the n_episode case, resetting in the beginning of `collect` is just fine. 
In the n_episode case, resetting is only happening the first time `collect` is called (which could be checked i.e. by checking for empty buffer). Do you prefer this?

Reply:
Empty buffer is not a reliable check, the user could have provided a filled buffer to the collector.

Instead, we could extend the signature of collect to contain `reset_env: bool = True`. The user passes the env and should be able to control whether it should be reset at the beginning of collect.

Note that the collector currently resets the env when it is instantiated, which is a different functionality. We should check for uses of the collector to see that we don't break anything when the resetting logic is moved to `.collect`.

After this has been done, the collector becomes essentially stateless, which will be a significant improvement.